src/query/exist_query.rs

```diff
@@ -0,0 +1,287 @@
+use core::fmt::Debug;
+
+use columnar::{ColumnType, DynamicColumn};
+
+use super::range_query::VecCursor;
+use super::ConstScorer;
+use crate::core::SegmentReader;
+use crate::docset::{DocSet, TERMINATED};
+use crate::query::explanation::does_not_match;
+use crate::query::{EnableScoring, Explanation, Query, Scorer, Weight};
+use crate::schema::FieldType;
+use crate::{DocId, Score, TantivyError};
+
+/// Query that matches all of the documents.
+///
+/// All of the document get the score 1.0.
+#[derive(Clone, Debug)]
+pub struct ExistsQuery {
+    field: String,
+}
+
+impl ExistsQuery {
+    /// Creates a new `RangeQuery` from bounded start and end terms.
+    ///
+    /// If the value type is not correct, something may go terribly wrong when
+    /// the `Weight` object is created.
+    pub fn new_exists_query(field: String) -> ExistsQuery {
+        ExistsQuery { field }
+    }
+}
+
+impl Query for ExistsQuery {
+    fn weight(&self, enable_scoring: EnableScoring<'_>) -> crate::Result<Box<dyn Weight>> {
+        let schema = enable_scoring.schema();
+        let field_type = schema
+            .get_field_entry(schema.get_field(&self.field)?)
+            .field_type();
+        if !field_type.is_fast() {
+            return Err(TantivyError::SchemaError(format!(
+                "Field {:?} is not a fast field.",
+                self.field
+            )));
+        }
+        Ok(Box::new(ExistsWeight {
+            field: self.field.to_string(),
+            field_type: field_type.clone(),
+        }))
+    }
+}
+
+/// Weight associated with the `ExistsQuery` query.
+pub struct ExistsWeight {
+    field: String,
+    field_type: FieldType,
+}
+
+impl ExistsWeight {}
+
+impl Weight for ExistsWeight {
+    fn scorer(&self, reader: &SegmentReader, boost: Score) -> crate::Result<Box<dyn Scorer>> {
+        let fast_field_reader = reader.fast_fields();
+        let column_type = match self.field_type {
+            FieldType::Str(_) => Some(ColumnType::Str),
+            FieldType::U64(_) => Some(ColumnType::U64),
+            FieldType::I64(_) => Some(ColumnType::I64),
+            FieldType::F64(_) => Some(ColumnType::F64),
+            FieldType::Bool(_) => Some(ColumnType::Bool),
+            FieldType::Date(_) => Some(ColumnType::DateTime),
+            FieldType::Bytes(_) => Some(ColumnType::Bytes),
+            FieldType::IpAddr(_) => Some(ColumnType::IpAddr),
+            _ => None,
+        }
+        .expect("Should be here");
+        if let Some(dynamic_column) =
+            fast_field_reader.dynamic_column_handle(&self.field, column_type)?
+        {
+            let docset = ExistsDocSet::new(dynamic_column.open()?);
+            return Ok(Box::new(ConstScorer::new(docset, boost)));
+        } else {
+            return Err(TantivyError::SchemaError(format!(
```

Comment:
I wonder if we should return an empty doc set in that case. Either way the behavior is something for which we need a unit test and a documentation.